### ISSUE TITLE
Timer created for wrong person?

### ISSUE THREAD

**Issue by sdishman** [Strategery-Inc/jira-harvest-time-tracker#43]
We noticed some very strange behavior today, and I can't quite explain it but I'm hoping that by sending along our findings to the development team, you'll have a better idea of what might have happened.

Here's the rundown:

1. Developer A created a Harvest Timer (from JIRA, obviously) for Project X.
2. Developer A realized that Project X was the wrong Project to choose, so the time record had to be adjusted.  Developer went to Harvest and deleted the timer.
3. Developer A refreshed the JIRA Issue page so that the "Stop Timer" button disappeared and the "Track Time" button appeared instead.
4. Developer A repeated step #1, instead using Project Y (the correct project).
5. After this was done, Developer **B** noticed that a time record was created under _their_ Harvest account, which was an exact duplicate of the one created in step #1.  However, Developer B never initiated this time entry - it just randomly showed up.  It has the exact same Project, Task, Description, and Start Time as the timer from step #1, but it's for the wrong user!

I'm wondering if the wires occasionally get crossed coming from JIRA, and the wrong user is sent to create a timer via the Harvest API?

I've included screenshots below of each developer's time records in Harvest that reflect the situation.  The screenshot for Developer A is after step 4, so it reflects the "new", correct timer (with the updated Project).  The screenshot for Developer B reflects the timer mistakenly created on their behalf somehow.

Developer A Timer (with corrected Project):

![developer a timer corrected](https://cloud.githubusercontent.com/assets/314529/7483965/aa66d32c-f351-11e4-8676-725ef801d752.png)

Developer B Timer (not sure where this time entry came from???)

![developer b timer bad](https://cloud.githubusercontent.com/assets/314529/7483970/ae966b10-f351-11e4-8eb0-86a45deda4ba.png)

**Comment by gabrielsomoza** [comment]
Hi @sdishman,

Are you able to consistently duplicate this? The Harvest session is stored into the users's JIRA session, which means that as long as each JIRA user has a different Harvest user then they should never mix up.

Could you confirm that both developers were using separate JIRA and also separate Harvest accounts?

**Comment by sdishman** [comment]
Not consistently reproducible (tried immediately afterwards following the same steps and couldn't reproduce).

The two developers have completely separate JIRA/Harvest users, and were working on completely separate machines.

**Comment by gabrielsomoza** [comment]
I'll close this ticket because we'll start work on refactoring session management. Please follow updates on the ticket mentioned above (#44).
